Source/Core/VideoCommon/GXPipelineTypes.h

```diff
@@ -39,7 +39,7 @@ struct GXPipelineUid
   GXPipelineUid(const GXPipelineUid& rhs) { std::memcpy(this, &rhs, sizeof(*this)); }
   GXPipelineUid& operator=(const GXPipelineUid& rhs)
   {
-    std::memcpy(this, &rhs, sizeof(*this));
+    std::memcpy(reinterpret_cast<void*>(this), &rhs, sizeof(*this));
     return *this;
   }
   bool operator<(const GXPipelineUid& rhs) const
```

Comment:
Do we want to add the reinterpret_cast for the memcmp as well?

Reply:
Compiler doesn't seem to care about what's being read from, only what's written to.

Reply:
The docs for -Wclass-memaccess explcitly state this too: "Warn when the destination of a call to a raw memory function such as memset or memcpy is an object of class type..."